packages/data-store/src/entities/contact/ContactMetadataItemEntity.ts

```diff
@@ -0,0 +1,50 @@
+import { Entity, Column, PrimaryGeneratedColumn, BaseEntity, ManyToOne, BeforeInsert, BeforeUpdate } from 'typeorm'
+import { ValidationConstraint } from '../../types'
+import { BaseContactEntity } from './BaseContactEntity'
+import { IsNotEmpty, validate, ValidationError } from 'class-validator'
+import { getConstraint } from '../../utils/ValidatorUtils'
+
+@Entity('ContactMetadata')
+export class ContactMetadataItemEntity extends BaseEntity {
+  @PrimaryGeneratedColumn('uuid')
+  id!: string
+
+  @Column({ name: 'label', length: 255, nullable: false })
+  @IsNotEmpty({ message: 'Blank metadata labels are not allowed' })
+  label!: string
+
+  @Column({ name: 'valueType', nullable: false })
+  @IsNotEmpty({ message: 'valueType must not be empty' })
```

Comment:
should we add a test for this, like we do for the other checks?

Reply:
same for the identity metadata

Reply:
I've added extra tests in , but not for blank valueType because from the API there is no code path do achieve this.

Reply:
you should be able too? we are doing that for all of these blank values? the code should check for blank values optherwise the user justs sees a sql error?

Reply:
You cannot do that, valueType is not present in the DTO and not reachable from the contactMetadataItemEntityFrom(metadataItem)